diagnostics/runoff_sensitivities/runoff_sensitivities_diag.py

```diff
@@ -0,0 +1,1019 @@
+# <DATADIR>/<frequency>/<CASENAME>.<variable_name>.<frequency>.nc
+# Here <variable_name> and frequency are requested in the "varlist" part of 
+# settings.json.
+
+# The following command sets input_path to the value of the shell environment
+# variable called TAS_FILE. This variable is set by the framework to let the 
+# script know where the locally downloaded copy of the data for this variable
+# (which we called "tas") is.
+
+## load model deta
+# data path from framework
+pr_path = os.environ["PR_FILE"]
+tas_path = os.environ["TAS_FILE"]
+hfls_path = os.environ["HFLS_FILE"]
+mrro_path = os.environ["MRRO_FILE"]
+
+# model grids (pr,tas,hfls,mrro must have same lon,lat grids)
+lat_coor_name = os.environ["lat_coord"]
+lon_coor_name = os.environ["lon_coord"]
+time_coord_name = os.environ["time_coord"]
+latm = nc.Dataset(pr_path).variables[lat_coor_name][:]
+lonm = nc.Dataset(pr_path).variables[lon_coor_name][:]
+time = nc.Dataset(pr_path).variables[time_coord_name][:]
+nlat = len(latm)
+nlon = len(lonm)
+nt = len(time)
+nyr = int(nt/12)
+syr_model = int(os.environ["FIRSTYR"])
+eyr_model = int(os.environ["LASTYR"])
+
+# variables
+pr_var_name = os.environ["pr_var"]
+tas_var_name = os.environ["tas_var"]
+hfls_var_name = os.environ["hfls_var"]
+mrro_var_name = os.environ["mrro_var"]
+prm = nc.Dataset(pr_path).variables[pr_var_name][:,:,:].filled(fill_value=np.nan)
+tasm = nc.Dataset(tas_path).variables[tas_var_name][:,:,:].filled(fill_value=np.nan)
+hflsm = nc.Dataset(hfls_path).variables[hfls_var_name][:,:,:].filled(fill_value=np.nan)
+mrrom = nc.Dataset(mrro_path).variables[mrro_var_name][:,:,:].filled(fill_value=np.nan)
+
+## change lon & variables to western degree, since the basin mask is not closed in eastern degree.
+if np.mean(lonm) > 100:
+    # change lon to western degree
+    lonw = lonm - 180
+    lonw[lonw < 0] = 1000
+    ind180 = np.argmin(lonw)
+    lonwm = np.concatenate((lonm[ind180:] - 360, lonm[0:ind180]))
+    western_ind=np.concatenate((range(ind180, nlon), range(0, ind180)))
+    prm = prm[:, :, western_ind]
+    tasm = tasm[:, :, western_ind]
+    mrrom = mrrom[:, :, western_ind]
+    hflsm = hflsm[:, :, western_ind]
+
+## water year average (Oct-Sep water year for every basin)
+## do time average and then do basin average (save computation time a lot)
+syr_sens=1945
+eyr_sens=2014
+nyr = eyr_sens - syr_sens
+inds=(syr_sens-syr_model)*12 + 10 - 1
+inde=inds + (nyr)*12
+prm_wy = np.nanmean(np.reshape(prm[inds:inde,:,:],(12,nyr,nlat,nlon),order="F"), axis=0)
+tasm_wy = np.nanmean(np.reshape(tasm[inds:inde,:,:],(12,nyr,nlat,nlon),order="F"), axis=0)
+mrrom_wy = np.nanmean(np.reshape(mrrom[inds:inde,:,:],(12,nyr,nlat,nlon),order="F"), axis=0)
+hflsm_wy = np.nanmean(np.reshape(hflsm[inds:inde,:,:],(12,nyr,nlat,nlon),order="F"), axis=0)
+
+## area average for specific river basins
+# load basin masks from shapefile provided by GRDC
+shapefile_path = "{OBS_DATA}/mrb_basins.shp".format(**os.environ)

```

Comment:
Please provide a tarball or link to a website or ftp server with required observational files to me and @bitterbark.